### ISSUE TITLE
C# Native libraries not copied to build output directory if Grpc metapackage is used (starting from v1.19)

### ISSUE THREAD

**Issue by yulia-d** [grpc/grpc#18824]
### What version of gRPC and what language are you using?
gRPC C#, issue started in 1.19
 
### What operating system (Linux, Windows,...) and version?
 Windows
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
VS 2017 

Create .NET Standard library and change target framework to net461, for example.
Add Grpc nuget in 1.19 or 1.20, build. Check output - it does not contain native assemblies.
For comparison, change version to 1.18 and build - build output will contain native assemblies.

**Comment by yulia-d** [comment]
Minimal example:
[ClassLibrary1.zip](https://github.com/grpc/grpc/files/3106285/ClassLibrary1.zip)

**Comment by jtattermusch** [comment]
It should be possible to track down the rootcause based on the .nupkg package contents (compare 1.19 nuget with 1.18 nuget).
Btw, what's interesting is that we have distribution tests that should reveal if the native libraries are not copied correctly.

**Comment by yulia-d** [comment]
I tried to compare the 1.18.0 and 1.19.0 folders in nuget cache and couldn't figure it out, but I don't have any experience in this area. Saw the targets file and it's identical in both versions. Need for someone with expertise to check it out...

I suspect the distribution tests don't catch it because they don't cover this special case. 
Note that it's not a standard .NET Framework project that would be created using VS .NET Framework console or library - it's a .NET SDK project that targets .NET Framework, as can be seen in the attached example.

The reason we can't simply work this around by switching to 'regular' project is that we actually have a dual-targeting .NET SDK library project that targets both net461 and netstandard2.0. The assemblies are missing from net461 build output and from the output of the regular .NET Framework application that consumes it.

**Comment by yulia-d** [comment]
BTW this is preventing us from incorporating the #18057 fix...

**Comment by yulia-d** [comment]
@jtattermusch  Any news about this? I see @johannesegger is also affected by it.

**Comment by jtattermusch** [comment]
Is this possibly a duplicate of https://github.com/grpc/grpc/issues/18737?

**Comment by jtattermusch** [comment]
I am able to reproduce with the example provided, but I'm not sure what's causing it. There seems to be no difference between https://github.com/grpc/grpc/commits/master/src/csharp/Grpc.Core/build/net45/Grpc.Core.targets in versions 1.19.0 and 1.18.0

**Comment by jtattermusch** [comment]
@yulia-d  I think I understand what's going on:
- your project is referencing the `Grpc` nuget package, which is a metapackage (currently depends on Grpc.Core only).
- between v1.18.0 and v1.19.0 we changed the definition of the `Grpc` metapackages in this refactoring PR: https://github.com/grpc/grpc/pull/17972
- after refactoring the `Grpc` metapackage, something doesn't behave in the same way as it used to (the behavior you observed) - I think the transitive inclusion of the .targets file got broken by the refactoring, so the .targets file needed by net45+ is not being included in the build and the native libraries are missing.
 
The good news is that the workaround is trivial - you can just reference Grpc.Core in your projects directly and everything will work again.

I will look into fixing the problem in `Grpc` nuget.

**Comment by yulia-d** [comment]
@jtattermusch Thanks for the great idea, this work-around unblocks us!
